libs/moment-datetime-adapter/adapter/moment-datetime-adapter.ts

```diff
@@ -0,0 +1,312 @@
+        return this.clone(date).day();
+    };
+
+    getHours(date: Moment): number {
+        return this.clone(date).hours();
+    };
+
+    getMinutes(date: Moment): number {
+        return this.clone(date).minutes();
+    };
+
+    getSeconds(date: Moment): number {
+        return this.clone(date).seconds();
+    };
+
+    setHours(date: Moment, hours: number): Moment {
+        return this.clone(date).hours(hours);
+    };
+
+    setMinutes(date: Moment, minutes: number): Moment {
+        return this.clone(date).minutes(minutes);
+    };
+
+    setSeconds(date: Moment, seconds: number): Moment {
+        return this.clone(date).seconds(seconds);
+    };
+
+    getWeekNumber(date: Moment): number {
+        return this.clone(date).week();
+    };
+
+    getMonthNames(style: 'long' | 'short' | 'narrow'): string[] {
+        switch (style) {
+            case 'narrow':
+                return this._localeData.narrowMonths;
+            case 'short':
+                return this._localeData.shortMonths;
+            case 'long':
+            default:
+                return this._localeData.longMonths;
+        }
+    };
+
+    getDateNames(): string[] {
+        return range(31, (i) => this.createDate(2017, 0, i + 1).format('D'));
+    };
+
+    getDayOfWeekNames(style: 'long' | 'short' | 'narrow'): string[] {
+        switch (style) {
+            case 'narrow':
+                return this._localeData.narrowDaysOfWeek;
+            case 'short':
+                return this._localeData.shortDaysOfWeek;
+            case 'long':
+            default:
+                return this._localeData.longDaysOfWeek;
+        }
+    };
+
+    getYearName(date: Moment): string {
+        return this.clone(date).format('YYYY');
+    };
+
+    getWeekName(date: Moment): string {
+        return this.clone(date).week().toLocaleString(this.locale);
+    };
+
+    getHourNames({ meridian, twoDigit }: { twoDigit: boolean; meridian: boolean }): string[] {
+        const format: string = meridian ? (twoDigit ? 'hh' : 'h') : (twoDigit ? 'HH' : 'H');
+        const momentDate = this._createMomentDate();
+
+        return range(24, (i) => {
+            return this.clone(momentDate).hour(i).format(format);
+        });
+    };
+
+    getMinuteNames({ twoDigit }: { twoDigit: boolean }): string[] {
+        const format: string = twoDigit ? 'mm' : 'm';
+        const momentDate = this._createMomentDate();
+
+        return range(60, (i) => {
+            return this.clone(momentDate).minute(i).format(format);
+        });
+    };
+
+    getSecondNames({ twoDigit }: { twoDigit: boolean }): string[] {
+        const format: string = twoDigit ? 'ss' : 's';
+        const momentDate = this._createMomentDate();
+
+        return range(60, (i) => {
+            return this.clone(momentDate).second(i).format(format);
+        });
+    };
+
+    getDayPeriodNames(): [string, string] {
+        console.log('here');
+
+        return ['', ''];
```

Comment:
We could try retrieve this from locale data

![image](https://user-images.githubusercontent.com/3263744/107204714-aaed6f00-6a05-11eb-95ac-b1a7e95a014d.png)

Reply:
Changed